transcom/mymove
# MAIN B-19164 add Okta customer group id for each environment to chamber & update config files (MERGE AFTER MAIN B-18821)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18916)
[INTEGRATION BRANCH PR](https://github.com/transcom/mymove/pull/12287)

This includes code from Dre's [MAIN B-18821 PR](https://github.com/transcom/mymove/pull/12281). Merge this after that PR gets merged.

## Summary

We will be adding functionality to add customers as an office user, which will have the option of creating Okta accounts for those users. In order to do that, we will need to ensure we add those users to the Okta groups which allow access to the customer application when redirected back to MilMove.

[this confluence page](https://dp3.atlassian.net/wiki/spaces/MT/pages/1249542242/0030+How+to+Manage+Secrets+with+Chamber) explains more about the approach used when adding values to chamber. Added the Okta group ids to the following chamber services:
`app-devlocal`
`app-demo`
`app-exp`
`app-loadtest`
`app-stg`
`app-prd`

Also added a flag for the customer group so the backend can access that value using viper.

### How to test

1. In your terminal, run `code ~/.zshrc`
2. Ensure your `AWS_PROFILE` is set to `transcom-gov-dev`
3. `source ~/.zshrc` to refresh your terminal with the new values
4. `chamber list app-devlocal`
5. You should see `okta_customer_group_id` in `chamber`

Now you can rerun this flow for the following profiles & services (if you have access)
`transcom-gov-milmove-exp` --> `chamber list app-exp`
`transcom-gov-milmove-demo` --> `chamber list app-demo`
`transcom-gov-milmove-loadtest` --> `chamber list app-loadtest`
`transcom-gov-milmove-stg` --> `chamber list app-stg`
`transcom-gov-milmove-prd` --> `chamber list app-prd` (most of us will not have access to this one)
